### PR TITLE
perf: split NixState into NixStoreMixin

### PR DESCRIPTION
**Pull request by aakropotkin**
Previously the _mixin_ `flox::NixState` provided a connection to a nix store, and an evaluator.
This meant that for every evaluator we created, we were creating redundant nix store connections.

This change split `flox::NixState` members related to store connections into `flox::NixStoreMixin`, and updates `PkgDbRegistryMixin` to use _only_ a single nix store connection for all registry inputs.

Misc Changes:
- Fixes some `clang-tidy` issues in modified files.

### CODE REVIEW

File: include/flox/pkgdb/input.hh [hunk 1]
```diff
@@ -274,7 +283,18 @@ struct PkgDbRegistryMixin : virtual public NixState {
     /** @return A list of systems to be scraped. */
     [[nodiscard]] virtual std::vector<std::string> & getSystems() = 0;
```
Comment (zmitchell) [inline include/flox/pkgdb/input.hh:284]: TIL about `nodiscard`. What is setting this equal to 0 doing?

File: include/flox/resolver/state.hh [hunk 2]
```diff
@@ -50,68 +41,50 @@ class ResolverState : protected pkgdb::PkgDbRegistryMixin {
...
   public:
 
-    ResolverState( const RegistryRaw             & registry
-                 , const pkgdb::QueryPreferences & preferences
+    ResolverState( RegistryRaw             registry
+                 , pkgdb::QueryPreferences preferences
                  )
-      : registryRaw( registry )
-      , preferences( preferences )
+      : registryRaw( std::move( registry ) )
```
Comment (zmitchell) [inline include/flox/resolver/state.hh:63]: Was this previously doing a copy instead of a move?

File: include/flox/pkgdb/input.hh (hunk 1 above)
Comment (aakropotkin) [reply include/flox/pkgdb/input.hh:285]: it indicates that the function is "pure virtual", which means it is not implemented by _this_ class, and needs to be implemented in children.

File: include/flox/resolver/state.hh (hunk 2 above)
Comment (aakropotkin) [reply include/flox/resolver/state.hh:63]: Honestly it was just a `clang-tidy` recommendation, so in this case I get to link you to a nice blurb explaining :)
https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html

**Pull request merged**